### PR TITLE
ImagesService original -> thumb -> original

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/platform-team @drsnyder 
https://wikia-inc.atlassian.net/browse/PLATFORM-695

More functions that depend on the string structure of thumbnails :(

### CODE REVIEW

File: includes/wikia/services/ImagesService.class.php [hunk 1]
```diff
@@ -157,27 +157,41 @@ private static function isDataTagImage($url) {
...
+				if ( strpos( $destSize, "px" ) !== false ) {
+					list( $width, $_ ) = explode( "px", $destSize );
+					$generator->scaleToWidth( $width );
+				} else {
+					list( $width, $height ) = explode( "x", $destSize );
+					$generator
+						->fixedAspectRatio()
+						->height( $height )
+						->width( $width );
+				}
```
Comment (drsnyder) [inline includes/wikia/services/ImagesService.class.php]: This is getting dicey with the backwards compatibility of the `$destSize` param. It accepts things like `200px` or `200x200`. I would favor breaking the vignette block out into a separate function that handles this behavior in a more sane way. Something like:

```php
public static function getThumbUrlFromFileUrl($imageUrl, $destSize, $newExtension = null) {
...
   if ( VignetteRequest::isVignetteUrl( $imageUrl ) ) {
      list($width, $height) = self::parseGetThumbUrlFromFileUrlDestSize($destSize);
      $url = self::getThumbUrlFromVignetteUrl($imageUrl, $width, $height);
   }
}
```

This way, the logic should be easier to follow, test, and allow deprecation and removal of the old stuff.

Comment (nmonterroso) [inline includes/wikia/services/ImagesService.class.php]: sure

File: includes/wikia/services/ImagesService.class.php [hunk 2]
```diff
@@ -157,28 +188,30 @@ private static function isDataTagImage($url) {
...
+			if ( VignetteRequest::isVignetteUrl( $imageUrl ) ) {
+				list( $width, $height ) = self::parseThumbDestSize( $destSize );
+				$imageUrl = self::vignetteOriginalToThumb( $imageUrl, $width, $height );
 			} else {
-				$imageUrl = $imageUrl;
-			}
-
+				if ( !self::IsExternalThumbnailUrl( $imageUrl ) ) {
+					$imageUrl = str_replace( '/images/', '/images/thumb/', $imageUrl );
+				}
```
Comment (drsnyder) [inline includes/wikia/services/ImagesService.class.php]: Is the indentation off here?

File: includes/wikia/services/ImagesService.class.php [hunk 3]
```diff
@@ -157,28 +188,30 @@ private static function isDataTagImage($url) {
...
+				 *
+				 * - get last part of path
+				 * - add it as thumbnail file prefixed with widthpx
+				 */
+				$parts = explode( "/", $imageUrl );
+				$file = array_pop( $parts );
+
+				if ( ctype_digit( (string)$destSize ) ) {
+					$destSize .= 'px';
+				}
```
Comment (drsnyder) [inline includes/wikia/services/ImagesService.class.php]: Are ther extra spaces here?

**Comment by drsnyder** [comment]
Other than the spacing, lgtm.

File: includes/wikia/services/ImagesService.class.php (hunk 2 above)
Comment (nmonterroso) [inline includes/wikia/services/ImagesService.class.php]: Nah, this is now an if inside the else, so the indentation is correct.

File: includes/wikia/services/ImagesService.class.php (hunk 3 above)
Comment (nmonterroso) [inline includes/wikia/services/ImagesService.class.php]: Indentation is now one level deeper, since we're checking if the url is a vignette url.

**Pull request merged**